src/main/java/com/algorand/algosdk/v2/client/common/Query.java

```diff
@@ -0,0 +1,48 @@
+package com.algorand.algosdk.v2.client.common;
+
+public abstract class Query {
+
+	private Client client;
+	private String getOrPost;
+	protected QueryData qd;
+	
+	protected Query(Client client, String getOrPost) {
+		this.client = client;
+		this.getOrPost = getOrPost;
+		this.qd = new QueryData();
+	}
+
+	protected abstract QueryData getRequestString();
+
+	protected <T>Response<T> baseExecute() throws Exception {
+
+		QueryData qData = this.getRequestString();
+		com.squareup.okhttp.Response resp = this.client.executeCall(qData, getOrPost);
+		if (resp.isSuccessful()) {
+			return new Response<T>(resp.code(), null, resp.body().string());
+		} else {
+			return new Response<T>(resp.code(), resp.body().string(), null);
+		}
+	}
+
+	public String getRequestUrl(int port, String host) {
```

Comment:
can this be protected?

Reply:
I guess this is probably useful for testing.

Reply:
Yes. it is used from the testing tools.